### DISCUSSION TITLE
建议：对浴霸类服务增加映射到climate型实体

### DISCUSSION THREAD

**Discussion by pcr** [XiaoMi/ha_xiaomi_home#466] [Ideas]
如题，我查看了代码发现climate.py中目前已经有了AirConditioner和Heater两种实现。
理论上浴霸也是Heater的一种。但映射规则中目前只有heater，没有ptc-bath-heater。
这种情况下HA中操作浴霸到没问题，但对浴霸实体进行显示时就比较不友好了。

由于对代码还没有深度了解，感觉bath-heater和Heater的实现还不能完全适配，因此没有贸然尝试将ptc-bath-heater映射到Heater。
希望这里能有所改进。

**Comment by SusanPhevos** [comment]
请帮忙验证一下 #470 功能是否正常

**Reply by pcr** [reply]
好的，我粗略看了下代码，应该是可以的，但由于#470还没Merge到主干，我验证起来要单独搭个环境才好操作。
另外，代码中有这样一段：
`                    elif item['name'].lower() in {'fan', 'ventilate'}:`
`                        self._hvac_mode_map[item['value']] = HVACMode.FAN_ONLY`
这会导致抽风和吹气变成同一个状态，大多SPEC的ventilate>fan,所以遍历list后fan实际就没了。
考虑到HVACMode中没有ventilate这个状态，我个人建议在prop.value_list同时具有fan和ventilate时，将ventilate映射为HVACMode.HEAT_COOL。这样虽然不完美，但不会导致功能缩减。

**Reply by pcr** [reply]
实际测试了一下，那段我说的代码确实出现了问题，但和我想象中的结果不太一样。
目前的效果是模式中有两个FAN_ONLY，并且两个选项的结果都是吹风，抽湿功能没有。
我尝试按我说的方式修改后，效果是可用的。
